### PR TITLE
Feature/shared model get account

### PR DESCRIPTION
**Pull request by x3medima17**
Added Query to shared model.
Implemented GetAccount query.

### CODE REVIEW

File: shared_model/backend/protobuf/queries/proto_query.hpp [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+
+namespace shared_model {
+  namespace proto {
+    class Query final : public CopyableProto<interface::Query,
+                                             iroha::protocol::Query,
+                                             Query> {
+     private:
+      /// polymorphic wrapper type shortcut
+      template <typename Value>
+      using w = detail::PolymorphicWrapper<Value>;
```
Comment (l4l) [inline shared_model/backend/protobuf/queries/proto_query.hpp]: It's better to use [this](https://github.com/hyperledger/iroha/blob/e42fa486452a7d6ba8def8f39981011236cfc27b/shared_model/backend/protobuf/transaction_responses/proto_tx_response.hpp#L53)

File: shared_model/backend/protobuf/queries/proto_query.hpp [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+
+      template <typename QueryType>
+      explicit Query(QueryType &&query)
+          : CopyableProto(std::forward<QueryType>(query)),
+            variant_(
+                [this] { return load_query<ProtoQueryListType>(*proto_); }),
+            blob_([this] { return BlobType(proto_->SerializeAsString()); }),
+            signatures_([this] {
+              SignatureSetType set;
+              SignatureType sig(new Signature(proto_->signature()));
```
Comment (l4l) [inline shared_model/backend/protobuf/queries/proto_query.hpp]: Why new?

File: shared_model/backend/protobuf/queries/proto_query.hpp [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+              set.emplace(sig);
```
Comment (l4l) [inline shared_model/backend/protobuf/queries/proto_query.hpp]: Either use insert, or call with the `Type(proto_->signature())`

File: test/module/shared_model/backend_proto/CMakeLists.txt [hunk 4]
```diff
@@ -16,11 +16,20 @@
...
+        )
+
+addtest(shared_proto_queries_test
+        shared_proto_queries_test.cpp
+        )
+
 target_link_libraries(shared_proto_commands_test
-    shared_model_proto_backend
-    )
+        shared_model_proto_backend
```
Comment (l4l) [inline test/module/shared_model/backend_proto/CMakeLists.txt]: Imo it's better group by target, like:
```
addtest(..)
link(..)
addtest(..)
link(..)
```

File: test/module/shared_model/backend_proto/shared_proto_queries_test.cpp [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+ * @when create shared model query object
+ * @then corresponding shared model object is created
+ */
+
+TEST(ProtoQuery, QueryLoad) {
+  iroha::protocol::Query query;
+  auto refl = query.GetReflection();
+  auto desc = query.GetDescriptor();
+  boost::for_each(
+      // TODO 11/27/17 andrei PR #695 replace 1 with desc->field_count()
```
Comment (l4l) [inline test/module/shared_model/backend_proto/shared_proto_queries_test.cpp]: D: there's a trap

File: shared_model/backend/protobuf/queries/proto_query.hpp (hunk 3 above)
Comment (x3medima17) [reply shared_model/backend/protobuf/queries/proto_query.hpp]: The constructor accepts only pointers

File: shared_model/backend/protobuf/queries/proto_query.hpp [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+      }
+
+      const BlobType &blob() const override { return *blob_; }
+
+      // ------------------------| Signable override  |-------------------------
+      const SignatureSetType &signatures() const override {
+        return *signatures_;
+      }
+
+      bool addSignature(const SignatureType &signature) override {
```
Comment (muratovv) [inline shared_model/backend/protobuf/queries/proto_query.hpp]: This method should return false always.

File: shared_model/backend/protobuf/queries/proto_get_account.hpp [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+    class GetAccount final : public CopyableProto<interface::GetAccount,
+                                                  iroha::protocol::Query,
+                                                  GetAccount> {
+     public:
+      template <typename QueryType>
+      explicit GetAccount(QueryType &&query)
+          : CopyableProto(std::forward<QueryType>(query)),
+
+            account_id_(detail::makeReferenceGenerator(
+                &proto_->payload(),
```
[the remark this answers is not in the document]
Comment (x3medima17) [reply shared_model/backend/protobuf/queries/proto_get_account.hpp]: Yes, it does

**Pull request merged**